client/homescreen/activity-panel/orders/index.js

```diff
@@ -304,8 +304,8 @@ export default withSelect( ( select, props ) => {
 		REPORTS_STORE_NAME
 	);
 
-	if ( ! countUnreadOrders ) {
-		return { isRequesting: countUnreadOrders !== 0 };
+	if ( ! orderStatuses.length && countUnreadOrders === 0 ) {
```

Comment:
Curious why we separate this logic from the condition below.  It's a little hard to follow, but it seems both are related to whether or not we're still requesting.

Reply:
If we know that at least one `order status` was selected and there aren't orders to show, it's not necessary to continue with the other requests that we do later in the code, [here](https://github.com/woocommerce/woocommerce-admin/blob/b0600d7071ea0f333cff7dc969b8e57297e7b572/client/homescreen/activity-panel/orders/index.js#L319) and [here](https://github.com/woocommerce/woocommerce-admin/blob/b0600d7071ea0f333cff7dc969b8e57297e7b572/client/homescreen/activity-panel/orders/index.js#L355).
So in order to have a more efficient behavior, I added this check before doing those calls to avoid continuing when we already have all the data that we need.
But for sure the logic is a little confusing and a future refactor would be a good idea.

Reply:
>But for sure the logic is a little confusing and a future refactor would be a good idea.

➕ 1️⃣